client/src/components/FormBuilderModal/FormBuilderModal.js

```diff
@@ -108,107 +103,80 @@ class FormBuilderModal extends Component {
    * @param {Function} submitFn The original submit function
    * @returns {Promise}
    */
-  handleSubmit(data, action, submitFn) {
-    this.clearResponse();
+  const handleSubmit = (data, action, submitFn) => {
+    clearResponse();
     let promise = null;
-    if (typeof this.props.onSubmit === 'function') {
-      promise = this.props.onSubmit(data, action, submitFn);
+    if (typeof onSubmit === 'function') {
+      promise = onSubmit(data, action, submitFn);
     } else {
       promise = submitFn();
     }
 
     if (promise) {
       // do not want this as part of the main promise chain.
       promise
-        .then((response) => {
-          if (response) {
-            this.setState({
-              response: response.message,
-              error: false,
-            });
+        .then((successResponse) => {
+          if (successResponse) {
+            setSuccess(successResponse.message);
           }
-          return response;
+          return successResponse;
         })
         .catch((errorPromise) => {
           errorPromise.then((errorText) => {
-            this.setState({
-              response: errorText,
-              error: true,
-            });
+            setFailure(errorText);
           });
         });
     } else {
       throw new Error('Promise was not returned for submitting');
     }
 
     return promise;
-  }
-
-  renderHeader() {
-    let { title } = this.props;
-    const { ModalHeaderComponent } = this.props;
-
-    if (title !== false) {
-      if (typeof title === 'object') {
-        // FormSchema title occasionaly contains html, only render text for modal title
-        const doc = new DOMParser().parseFromString(title.html, 'text/html');
-        title = doc.body.textContent || '';
+  };
+
+  const modalProps = {
+    className,
+    isOpen,
+    modalClassName,
+    ModalComponent,
+    ModalHeaderComponent,
+    onClosed: handleHide,
+    showCloseButton,
+    size,
+    title,
+
+  };
+  const formBuilderProps = {
```

Comment:
```suggestion
  const formBuilderLoaderProps = {
```
Make it consistent with the name of the component it's going into

Reply:
Done